src/hardware/siglent-sds/protocol.c

```diff
@@ -315,13 +336,25 @@ static int siglent_sds_read_header(struct sr_dev_inst *sdi)
 	struct dev_context *devc = sdi->priv;
 	char *buf = (char *)devc->buffer;
 	int ret, desc_length;
-	int block_offset = 15; /* Offset for descriptor block. */
+	int block_offset; /* Offset for descriptor block. */
+	if (!strcmp(devc->model->series->name, "SDS2000X+"))
```

Comment:
maybe would make sense to have this value (and header_size just below) in the device descriptor struct. Less code.